### DISCUSSION TITLE
Send Object through the Redis Stream Xadd

### DISCUSSION THREAD

**Discussion by SidathWeerasinghe** [redis/lettuce#2069] [Q&A] [answered]
When I send the Java Object using the below code it returns the below error. 
`Exception in thread "main" java.lang.IllegalArgumentException: Message body.length must be a multiple of 2 and contain a sequence of field1, value1, field2, value2, fieldN, valueN
	at io.lettuce.core.internal.LettuceAssert.isTrue(LettuceAssert.java:208)
	at io.lettuce.core.RedisCommandBuilder.xadd(RedisCommandBuilder.java:2622)
	at io.lettuce.core.AbstractRedisAsyncCommands.xadd(AbstractRedisAsyncCommands.java:1864)`

Code ->

` EventStructure eventStructure = new EventStructure(); 
   eventStructure.setPublisher("sidath");
            String messageId = syncCommands.xadd(
                    STREAMS_KEY,
                    eventStructure);
`

**Comment by mp911de** [comment, marked as answer]
Lettuce doesn't include an object mapper, therefore you need to pass key-value pairs or a `Map` into the `XADD` command.
